typing/typecore.ml

```diff
@@ -908,6 +908,31 @@ let type_continuation_pat env expected_ty sp =
           Types.val_loc = loc; val_attributes = []; }
       in
         Some (id, desc)
+  | Ppat_constraint ({ppat_desc=ddesc; ppat_loc=lloc},
+                     sty) ->
+      let separate = true in
+      if separate then begin_def();
+      let cty, force = Typetexp.transl_simple_type_delayed env sty in
+      let ty = cty.ctyp_type in
+      let ty, expected_ty' =
+        if separate then begin
+          end_def();
+          generalize_structure ty;
+          instance env ty, instance env ty
+        end else ty, ty
+      in
+      unify_pat_types lloc env ty expected_ty;
+      pattern_force := force :: !pattern_force;
+      (match ddesc with
```

Comment:
Instead of a nested match, would it be better for this to be a recursive call to `type_continuation_pat`?